uint16 left_limit
uint16 right_limit
